engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/compiler/TestFlinkProcessCompiler.scala

```diff
@@ -1,39 +1,82 @@
+import pl.touk.nussknacker.engine.graph.expression.Expression
 import pl.touk.nussknacker.engine.process.exception.FlinkExceptionHandler
+import pl.touk.nussknacker.engine.spel.SpelExpressionParser
 import pl.touk.nussknacker.engine.testmode.ResultsCollectingListener
+import pl.touk.nussknacker.engine.variables.GlobalVariablesPreparer
 
-class TestFlinkProcessCompiler(creator: ProcessConfigCreator,
-                               inputConfigDuringExecution: Config,
+class TestFlinkProcessCompiler(modelData: ModelData,
                                collectingListener: ResultsCollectingListener,
                                process: CanonicalProcess,
-                               scenarioTestData: ScenarioTestData,
-                               objectNaming: ObjectNaming)
-  extends StubbedFlinkProcessCompiler(process, creator, inputConfigDuringExecution, diskStateBackendSupport = false, objectNaming, ComponentUseCase.TestRuntime) {
+                               scenarioTestData: ScenarioTestData)
+  extends StubbedFlinkProcessCompiler(process, modelData.configCreator, modelData.processConfig, diskStateBackendSupport = false, modelData.objectNaming, ComponentUseCase.TestRuntime) {
 
-  
   override protected def adjustListeners(defaults: List[ProcessListener], processObjectDependencies: ProcessObjectDependencies): List[ProcessListener] = {
     collectingListener :: defaults
   }
 
+  private val expressionEvaluator: (Expression, Parameter, NodeId) => ValidatedNel[PartSubGraphCompilationError, AnyRef] = {
+    val validationContext = GlobalVariablesPreparer(modelData.processWithObjectsDefinition.expressionConfig).emptyValidationContext(process.metaData)
+    val evaluator = ExpressionEvaluator.unOptimizedEvaluator(modelData)
+    val dumbContext = Context("dumb", Map.empty, None)
+    val expressionCompiler = ExpressionCompiler.withoutOptimization(modelData).withExpressionParsers {
+      case spel: SpelExpressionParser => spel.typingDictLabels
+    }
+    (expression: Expression, parameter: Parameter, nodeId: NodeId) => {
+      expressionCompiler
+        .compile(expression, Some(parameter.name), validationContext, parameter.typ)(nodeId)
+        .map { typedExpression =>
+          val param = evaluatedparam.Parameter(typedExpression, parameter)
+          evaluator.evaluateParameter(param, dumbContext)(nodeId, process.metaData).value
+        }
+    }
+  }
+
+  private def collectSamples(originalSource: Any, sourceWithTestSupport: FlinkSourceTestSupport[Object@unchecked], nodeId: NodeId): List[Object] = {
+    val scenarioJsonRecords = scenarioTestData.testRecords.collect { case testRecord: ScenarioTestJsonRecord => testRecord }.filter(_.sourceId == nodeId)
+    val scenarioParamRecords = scenarioTestData.testRecords.collect { case testRecord: ScenarioTestParametersRecord => testRecord }.filter(_.sourceId == nodeId)
+
+    if(scenarioJsonRecords.nonEmpty && scenarioParamRecords.nonEmpty)
```

Comment:
Wouldn't pattern matching over `scenarioJsonRecords` and `scenarioParamRecords` be more readable?

Reply:
Yeah, I have the same notice. Let's rewrite it to pattern matching. Also why it is impossible to handle both testrecords types? In Lite implementation it works. It looks like the only thing to do is to change logic to dispatch handling logic for each record instead of doing it up-front for all records

Reply:
I've changed it to pattern matching + removed error, we can easily handle both testrecords if situation like this ever happens (don't thinks so). :+1:

Reply:
I'm moving flink implementation to separate PR so I'm closing this thread.